pkg/controller/mizar/mizar-service-controller.go

```diff
@@ -213,115 +214,143 @@ func (c *MizarServiceController) syncService(eventKeyWithType KeyWithEventType)
 	}
 
+	// update service cluster ip with ip from mizar
 	if len(service.Spec.ClusterIP) == 0 {
 		svcToUpdate := service.DeepCopy()
 		svcToUpdate.Spec.ClusterIP = ip
-		svcUpdated, err := c.kubeClientset.CoreV1().ServicesWithMultiTenancy(service.Namespace, service.Tenant).Update(svcToUpdate)
+		_, err := c.kubeClientset.CoreV1().ServicesWithMultiTenancy(service.Namespace, service.Tenant).Update(svcToUpdate)
 		if err != nil {
-			klog.Errorf("The following service failed to update: %v", svcUpdated)
+			klog.Errorf("Failed to update service %v/%v/%v: %v", service.Tenant, service.Namespace, service.Name, err)
 			return err
 		}
-		klog.Infof("Updated service: %v", svcUpdated)
 	}
 
-	if hasDNSServiceLabel && len(netName) != 0 {
-		klog.Info("[Mizar network controller] Arktos Network update starts ...")
-		net, err := c.netClient.ArktosV1().NetworksWithMultiTenancy(service.Tenant).Get(netName, metav1.GetOptions{})
+	// If this is the tenant dns service, update arktos network object with cluster ip from mizar
+	_, hasDNSServiceLabel := service.Labels[arktosapisv1.NetworkLabel]
+	if hasDNSServiceLabel {
+		networkObj, err := c.networkLister.NetworksWithMultiTenancy(service.Tenant).Get(defaultNetworkName)
 		if err != nil {
-			klog.Errorf("The following network failed to get: %v", net)
+			klog.Errorf("Failed to get network object %v/%v from informer. Error: %v ", service.Tenant, defaultNetworkName, err)
 			return err
 		}
-		if len(net.Status.DNSServiceIP) == 0 {
-			netReady := net.DeepCopy()
-			netReady.Status.DNSServiceIP = ip
-			netNew, err := c.netClient.ArktosV1().NetworksWithMultiTenancy(net.Tenant).UpdateStatus(netReady)
+		if len(networkObj.Status.DNSServiceIP) == 0 {
+			networkObjToUpdate := networkObj.DeepCopy()
+			networkObjToUpdate.Status.DNSServiceIP = ip
+			_, err := c.netClient.ArktosV1().NetworksWithMultiTenancy(networkObj.Tenant).UpdateStatus(networkObjToUpdate)
 			if err != nil {
-				klog.Errorf("The following network failed to update: %v", netReady)
+				klog.Errorf("Failed to update arktos network object %v/%v. Error: %v", networkObjToUpdate.Tenant, networkObjToUpdate.Name, err)
 				return err
 			}
-			klog.Infof("Updated network: %v", netNew)
 		}
 	}
 	return nil
 }
 
 func (c *MizarServiceController) processServiceUpdate(service *v1.Service, eventKeyWithType KeyWithEventType) error {
 	key := eventKeyWithType.Key
-	klog.Infof("processServiceUpdate network name is %v", service.Name)
+	klog.V(4).Infof("processServiceUpdate network name is %v", service.Name)
 	msg := &BuiltinsServiceMessage{
 		Name:          service.Name,
-		ArktosNetwork: "",
 		Namespace:     service.Namespace,
 		Tenant:        service.Tenant,
 		Ip:            service.Spec.ClusterIP,
+		Vpc:           service.Annotations[mizarAnnotationsVpcKey],
+		Subnet:        service.Annotations[mizarAnnotationsSubnetKey],
 	}
 	response := c.grpcAdaptor.UpdateService(c.grpcHost, msg)
 	code := response.Code
 
 	switch code {
 	case CodeType_OK:
```

Comment:
so the switch default to the function return, which is nil ?

Reply:
You are right. We never touch the logic of service update as we think it is not necessary. I will check with Phu and Hong further to see how we want to handle service update.